rust/protocol/src/sealed_sender.rs

```diff
@@ -757,31 +757,43 @@ pub async fn sealed_sender_encrypt_from_usmc<R: Rng + CryptoRng>(
     Ok(serialized)
 }
 
-/// This is a single-key multi-recipient KEM, defined in Manuel Barbosa's
-/// ["Randomness Reuse: Extensions and Improvements"]. It uses the "Generic Construction" in `4.1`
-/// of that paper, instantiated with ElGamal encryption.
+/// This module implements a single-key multi-recipient KEM, defined in Manuel Barbosa's
+/// ["Randomness Reuse: Extensions and Improvements"].
+///
+/// The KEM scheme summarized below is performed when calling
+/// [sealed_sender_multi_recipient_encrypt]. It uses the "Generic Construction" in `4.1` of that
+/// paper, instantiated with [ElGamal encryption].
+///
+/// This process is inverted by calling [sealed_sender_decrypt_to_usmc] with a *single* ciphertext
+/// from the set produced all at once by [sealed_sender_multi_recipient_encrypt]. In production,
+/// this extraction of single ciphertexts is performed implicitly by the Signal backend server,
+/// which then uses the additional metadata added in [sealed_sender_multi_recipient_encrypt] to
+/// securely route each message to the intended recipient. In testing,
+/// [sealed_sender_multi_recipient_fan_out] can be used to convert such a bulk message into
+/// a sequence of encrypted [UnidentifiedSenderMessageContent].
 ///
 /// ["Randomness Reuse: Extensions and Improvements"]: https://haslab.uminho.pt/mbb/files/reuse.pdf
+/// [ElGamal encryption]: https://en.wikipedia.org/wiki/ElGamal_encryption
 ///
 /// ### Overview
-/// 1. generate a random [sealed_sender_v2::DerivedKeys].
-/// 2. encrypt the keypair using a shared secret derived from the sender's private key and
-///    recipient's public key (this secret never changes for that pair of identity keys) with
-///    [sealed_sender_v2::apply_agreement_xor].
-/// 3. sign the encrypted keypair using the sender's private key with
-///    [sealed_sender_v2::compute_authentication_tag].
-/// 4. use the keypair from (1) to symmetrically encrypt the underlying
+/// The method described in `4.1` of ["Randomness Reuse: Extensions and Improvements"] enables
+/// reusing a single sequence of random bytes across multiple messages with the same content. This
+/// reduces computation time for clients sending the same message to multiple recipients without
+/// compromising the message security. The high-level steps of this process are listed below:
+/// 1. Generate a series of random bytes.
+/// 2. Derive an ephemeral key pair [sealed_sender_v2::DerivedKeys] from (1).
+/// 3. Encrypt (1) using a shared secret derived from the private ephemeral key (2) and the
+///    recipient's public identity key with [sealed_sender_v2::apply_agreement_xor].
+/// 4. Add an authentication tag with [sealed_sender_v2::compute_authentication_tag] for (3) using
+///    a secret derived from the sender's private identity key and the recipient's public
+///    identity key.
+/// 5. Use the keypair from (2) to symmetrically encrypt the underlying
 ///    [UnidentifiedSenderMessageContent].
```

Comment:
Yeah, I think I got to that realization as well. I tried deleting this numbered list section and I *think* I still prefer having it, but we can probably afford to be more concise since the pseudocode + paper link is actually extremely useful.